### ISSUE TITLE
Triangle channel type does not compute

### ISSUE THREAD

**Issue by jandevaty** [storm-fsv-cvut/smoderp2d#151]
q365 in stream_shape table is 0 which leads to following runtime error.
Will probably be the same for every profile shape with bottom width of 0 (zero) because then th virtual bottom width is calculated from q365 ...

```py
File "D:\Dokumenty\SMODERP\smoderp2d\smoderp2d\core\stream.py", line 104, in stream_reach_outflow
    r.outflow_method(r, dt)
  File "D:\Dokumenty\SMODERP\smoderp2d\smoderp2d\stream_functions\stream_f.py", line 157, in triangle
    he = compute_h(A=Ve / reach.length, m=reach.m, b=B)  # funkce pouzita pro lichobeznik  ____
  File "D:\Dokumenty\SMODERP\smoderp2d\smoderp2d\stream_functions\stream_f.py", line 36, in compute_h
    h_pre = A / b
ZeroDivisionError: float division by zero
```

**Comment by landam** [comment]
See https://github.com/storm-fsv-cvut/smoderp2d/blob/jandevaty-patch-1/smoderp2d/stream_functions/stream_f.py#L141

**Comment by landam** [comment]
@pesekon2 Any progress?

**Comment by pesekon2** [comment]
> @pesekon2 Any progress?

I think it got solved in 0cdfa4f. Right, @kavkapet ?

**Comment by kavkapet** [comment]
this is still problem: 
![image](https://user-images.githubusercontent.com/26659753/234000897-60683ac3-26a2-4bc1-b445-95c8b4c7ab7e.png)
 @jandevaty you are looking on it?

when smoderp start without stream, fall here:
![image](https://user-images.githubusercontent.com/26659753/234001263-8b620899-92f6-4ee2-b04a-2203b1c4906b.png)

**Comment by pesekon2** [comment]
I know where the problem is now. Gimme a second.

**Comment by pesekon2** [comment]
@landam:  Try it now.

**Comment by kavkapet** [comment]
move this issue up :)
![image](https://github.com/storm-fsv-cvut/smoderp2d/assets/26659753/d1c4e794-f9d8-4486-8fca-f3e2c42077c3)

**Comment by kavkapet** [comment]
![image](https://github.com/storm-fsv-cvut/smoderp2d/assets/26659753/cd2196a7-c0e2-4718-b006-53668935ee27)

**Comment by kavkapet** [comment]
why?

**Comment by pesekon2** [comment]
@kavkapet: The problem was not connected to the `triangle` channel shape type (your error message was reached in the `trapezoid` part). It was therefore not connected to this issue but was an issue indeed. 

The reason was wrong parameter handling in `Logger.error`. I have just fixed it in 8e362dd.

**Comment by pesekon2** [comment]
None of the problems mentioned here is connected to the `triangle` channel type. Closing the issue.
